src/uct/ib/rc/accel/rc_mlx5.inl

```diff
@@ -1601,8 +1696,14 @@ uct_rc_mlx5_common_dm_make_data(uct_rc_mlx5_iface_common_t *iface,
         ucs_assert(desc->super.buffer != NULL);
         buffer = (void*)UCS_PTR_BYTE_DIFF(iface->dm.dm->start_va, desc->super.buffer);
 
-        uct_rc_mlx5_iface_common_copy_to_dm(cache, hdr_len, payload,
-                                            length, desc->super.buffer, log_sge);
+        if (iovcnt > 0) {
+            uct_rc_mlx5_iface_copy_iov_to_dm(cache, hdr_len, iov, iovcnt,
+                                             desc->super.buffer, log_sge);
+        } else {
+            uct_rc_mlx5_iface_common_copy_to_dm(cache, hdr_len, payload, length,
+                                                desc->super.buffer, log_sge);
```

Comment:
can ww remove uct_rc_mlx5_iface_common_copy_to_dm and use 
iov.buffer=payload
iov.length=length
uct_rc_mlx5_iface_copy_iov_to_dm(... &iov, 1)?

Reply:
We can. After that it will be possible to remove payload and length parameters from uct_rc_mlx5_common_dm_make_data, remove uct_rc_mlx5_ep_short_iov_dm method.
But we will add extra work to existing methods creating iov.
I can try and check if it will hurt performance.

Reply:
yep, let's see if compiler is smart, if not let's keep as is